app/uk/gov/hmrc/apipublisher/connectors/MicroserviceConnector.scala

```diff
@@ -41,7 +50,17 @@ class MicroserviceConnector @Inject()(ramlLoader: RamlLoader, http: HttpClient)
 
   def getAPIAndScopes(serviceLocation: ServiceLocation)(implicit hc: HeaderCarrier): Future[Option[ApiAndScopes]] = {
     val url = s"${serviceLocation.serviceUrl}/api/definition"
-    http.GET[Option[ApiAndScopes]](url) recover unprocessableRecovery
+    http.GET[Option[ApiAndScopes]](url).map(validateApiAndScopesAgainstSchema) recover unprocessableRecovery
+  }
+
+  private def validateApiAndScopesAgainstSchema(apiAndScopes: Option[ApiAndScopes]): Option[ApiAndScopes] = {
+    apiAndScopes map { definition =>
+      val inputStream: InputStream = env.resourceAsStream("api-definition-schema.json").get
+      val schema: Schema = SchemaLoader.load(new JSONObject(IOUtils.toString(inputStream, UTF_8)))
```

Comment:
Might it be preferable to just load the schema definition in once at construction time, rather than every time we want to perform a validation?

Reply:
Good point. It's always the same.